core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelFile.java

```diff
@@ -467,4 +478,23 @@ public CamelModel getCamelModel(){
 		return CamelModelFactory.getModelForProject(resource != null ? resource.getProject() : null);
 	}
 
+	@Override
+	public List<AbstractCamelModelElement> findAllNodesWithId(String nodeId) {
+		List<AbstractCamelModelElement> result = new ArrayList<>();
+
+		if (getGlobalDefinitions() != null && !getGlobalDefinitions().isEmpty()) {
```

Comment:
I think that the check for non-emptiness is not useful and can be removed to simplify the code.
getGlobalDefinitions().values() will return an empty Collection and it will go fine.